src/metrics.rs

```diff
@@ -2467,6 +2542,12 @@ impl GooseAttack {
         let mut received_message = false;
         let mut message = goose_attack_run_state.metrics_rx.try_recv();
 
+        let mut tasks = 0;
+        for set in self.task_sets.iter() {
+            tasks += set.tasks.len();
+        }
+        self.metrics.record_users_tasks_per_second(tasks as u32);
```

Comment:
> Why do we calculate this before merging in the received statistics? Shouldn't we merge statistics and then count?

I don't think this would change anything. This is simply counting tasks and users at this point and I don't think other stats would influence that. 

> Also, why does this happen here, instead of in the same place as where we record the other metrics for the graphs?

Because number of tasks do not need to be recorded for every request that we are collecting stats for; we need to do it just once.

Reply:
This still feels like an illogical place to perform this function, especially if we don't actually care whether or not we receive the metrics (the purpose of this function).

Reply:
Agreed. Moved it to `GooseAttack::sync_metrics()`. Will that work?